libs/core/langchain_core/utils/utils.py

```diff
@@ -38,6 +38,14 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
 
     return decorator
 
+T = TypeVar('T')
+H = TypeVar('H', bound=Hashable)
+def unique_by_key(iterable: Iterable[T], key: Callable[[T], H]) -> Iterator[T]:
```

Comment:
maybe just worth being local to ensemble retriever until we use in multiple places?

Reply:
Done!

Reply:
@hwchase17 Any more changes you'd like to see?

Reply:
@hwchase17 I apparently needed to run `make lint` at the langchain_core level as well (root didn't detect the lint fails on CI), it should be ok now.